.github/workflows/main.yml

```diff
@@ -32,6 +32,32 @@ jobs:
 
       - run: python mirror.py
 
-      - run: |
+      - name: check for unpushed commits
+        id: check_unpushed
+        run: |
+          UNPUSHED_COMMITS=$(git log origin/main..HEAD)
+          if [ -z "$UNPUSHED_COMMITS" ]; then
+            echo "No unpushed commits found."
+            echo "changes_exist=false" >> $GITHUB_ENV
+          else
+            echo "Unpushed commits found."
+            echo "changes_exist=true" >> $GITHUB_ENV
+          fi
```

Comment:
This seems fine but I wonder if `mirror.py` should indicate if this needs to happen or not? Perhaps for tracking in a follow-up

Reply:
The created variable allows for any extra function to be added based on `$changes_exist`. Making `mirror.py` send out some type of signal doesn't make the most sense to me. But maybe you have something specific in mind?

Reply:
Well it could be as simple as "echo "changes_exist=$(python mirror.py)" >> $GITHUB_ENV" and having `mirror.py` emit `true` or `false` to STDOUT. The idea here is that `mirror.py` would say if you need to publish a release instead of checking for extra commits and releasing based on a side-effect which is more likely to be error prone.

Reply:
I think I figured out a way to make this work. However, it is dependent on side effects. So let us say we changed mirror.py to look like:

https://github.com/CoderJoshDK/ruff-pre-commit/blob/a628536dcda0210e9733cf7056947769fc58e4b4/mirror.py#L12-L25

And then changed `main.yml` to look like:

https://github.com/CoderJoshDK/ruff-pre-commit/blob/a628536dcda0210e9733cf7056947769fc58e4b4/.github/workflows/main.yml#L33-L42

It would roughly do what you want. However, there are multiple issues with this approach. It works (should, will make sure to do more testing if you are ok with this issue) but only because of the side effect where the subprocess in mirror writes to STDOUT. And so now we end up trading one side effect for another. If you still feel that this is cleaner, that is fine, and I can put these changes in. 
Example output:
```sh
❯ echo $(python3 mirror.py)
[test f00a01e] Mirror: 0.3.5 1 file changed, 1 insertion(+), 1 deletion(-)


❯ echo $(python3 mirror.py)


```

And if you do want this change, do you want it as a separate PR?

Reply:
Ah I would not depend on the subprocess writing to stdout that's too confusing. Let's just consider this separately there's no reason to slow down this fix. I probably ought to actually look at it myself.

Reply:
You could always have it print out some random value when a commit is made, just to double make sure a write to the `STDOUT` is done. But the code ends up being exactly the same with an extra `print`.

Personally, I feel like the current state of things is easier to follow. Being too cleaver might not actually make anything better. Regardless, I am interested to see if you manage to think of a solution.